### ISSUE TITLE
Conflict with directives `seealso`, `deprecated`, ...

### ISSUE THREAD

**Issue by soxofaan** [tox-dev/sphinx-autodoc-typehints#467]
Minimal reproduction example:

```python

def hello() -> str:
    """
    .. seealso:: more info at <https://example.com>`_.
    """
    return "Hello, world!"
```

This triggers this warning during build:

> .../helloworld.py:docstring of helloworld.hello:3: WARNING: Explicit markup ends without a blank line; unexpected unindent.

(using sphinx==7.4.1, sphinx_autodoc_typehints==2.2.2)

**Comment by soxofaan** [comment]
when running with `-vv`, I see this output is generated:

```
[autodoc] output:

.. py:module:: helloworld


.. py:function:: hello()
   :module: helloworld


   .. seealso:: more info at <https://example.com>`_.
   :rtype: :sphinx_autodoc_typehints_type:`\:py\:class\:\`str\``
```

rst apparently expects a new line between the `seealso` directive and the `rtype` field


Workaround is to manually add a newline, like so:
```python
    """
    .. seealso:: more info at <https://example.com>`_.

    """
```
but that is of course not ideal as it is very tempting to clean up that trailing whitespace if you don't know what is for

**Comment by soxofaan** [comment]
comparable use case that triggers warning for same reason:

```python
def greetings() -> str:
    """
    .. deprecated:: 1.2.3
       Use :func:`hello` instead.
    """
    return "Hello, world!"
 ```

**Comment by hoodmane** [comment]
Thanks for the report @soxofaan, #468 should fix it.

**Comment by soxofaan** [comment]
awesone, I see you also released 2.2.3 including that fix :+1: 

I can confirm that this eliminates a bunch of warnings in our build
